src/test/suite/stubDemoTutorial.test.ts

```diff
@@ -97,6 +111,82 @@ suite('stub out a tutorial', () => {
 		}
 	}
 
+	async function validateTerminalResponseWin(terminalName : string, terminalText : string, terminalResponse : string[]) {
+		console.log(`validateTerminalResponse terminal ${terminalName} executing text ${terminalText} on a Windows machine`);
+		const term = window.createTerminal(terminalName);
+		expect(term).to.not.be.null;
+		if (term) {
+			console.log(`-current terminal = ${term?.name}`);
+			await sendTerminalText(terminalName, terminalText);
+			await waitUntil(async () => {
+				await focusOnNamedTerminal(terminalName);
+				return terminalName === window.activeTerminal?.name;
+			}, 1000);
+			try {
+				const terminalData = await collectTerminalStringsAsArray();
+				let foundExpectedString = false;
+				const responseArray = terminalResponse as string[];
+				console.log(`-terminal output =`);
+				responseArray.forEach(element => {
+					console.log(`---> ${element}`);
+				});
+				let foundIt = true;
+				for (let index = 0; index < responseArray.length; index++) {
+					const response = responseArray[index];
+					if (!response.startsWith(`PS`)) { // PowerShell lines begin with PS and we just want the others
+						const foundNum = searchStringInArray(response, terminalData);
+						foundIt = foundNum > -1;
+						if (!foundIt) {
+							break;
+						}
+					}
+				}
+				foundExpectedString = foundIt;
+
+				if (foundExpectedString) {
+					return;
+				} else {
+					fail(`Searching for ${terminalResponse} but not found in current content of active terminal ${window.activeTerminal?.name} : ${terminalData}`);
+				};
+			} catch (error){
+				fail(error);
+			} finally {
+				findAndDisposeTerminal(terminalName);
+			}
+		}	
+	}
+
+	async function collectTerminalStringsAsArray() : Promise<string[]> {
+		let terminalData:string[] = [];
+		const predicate = async () => {
+			const result: string = await getActiveTerminalOutput();
+			await commands.executeCommand('workbench.action.terminal.clear');
+			if (result.trim().length > 0) {
+				const split = splitLines(result.trim());
+				terminalData = terminalData.concat(split);
+			}
+			return true;
+		};
+		var numberOfTimes = TERMINAL_WAIT_RETRY_TIMES;
+		const delay = TERMINAL_WAIT_RETRY;
+		for (let i = 0; i < numberOfTimes; i++) {
+			await predicate();
+			await new Promise((res) => { setTimeout(res, delay); });
+		}
```

Comment:
why not using a waitUntil like in other places here?

Reply:
The example I found for testing a similar problem used this approach. waitUntil wasn't working for me with the predicate. This works.

Reply:
To clarify @apupier ... The issue with waitUntil is that it only runs until the predicate returns true. In this case, I want the predicate to be run 10 times to make sure that we give the terminal plenty of time to finish whatever activity may be running behind it. Though waitUntil is great for a "one and done" or "wait until this is complete once" - it doesn't handle the "run this 10 times and stop" case.